### PR TITLE
Allow more time for first frame - critical for static images to be reliably displayed

### PR DESCRIPTION
**Pull request by signumnova**
Have found from a lot of playing static images that sometimes they are not successfully presented and just leave a black screen.

Narrowed down to a line of code in the renderer that only allows 5mS for the lock on the device to become available otherwise the frame is dropped. Clearly there is the odd occasion where 5mS is not quite enough. This could be exacerbated by having multiple players running simultaneously where images are being opened on all players at the same time. (Which is what I was doing).

This PR increases the time to 100mS, but only for the very first frame.

### CODE REVIEW

**Comment by SuRGeoNix** [comment]
Hi @signumnova, this does not look right to me. I will need more information. Multiple players shouldn't affect this lock as it is not a static lock but per instance. I could understand a rendering delay because of multiple possible high resolution images but again, for each Player/Renderer should have only one present at the time. So, I'm not sure why you get a timeout there. Can you check the other thread that keeps the lock?

**Comment by signumnova** [comment]
Hi there - When I first encountered this, yes it was with higher resolution images.

It doesn't necessarily take multiple players to create the conditions for this directly - as you say the lock is not shared between players.  BUT, this mutex is also used by the VideoDecoder thread ... there is plenty of stuff going on there with the mutex held that is out of flyleaf's control timing wise. So in theory I would say there is something going on here that takes slightly longer than expected in that thread... and maybe that's where having multiple players may be an exacerbating factor.

Regardless as the code stands, with a video who cares if you lose the odd frame because of this (except I note that although this drop is logged at Debug level it it not included in drop count?). However for a static image losing that first frame is totally make or break.

In summary, with this particular one, I believe this is a "harmless" fix as it is written. I can't see any downside to it because any performance effect of it is confined to rendering the first frame only.

Cheers SN

**Comment by SuRGeoNix** [comment]
I wasn't able to reproduce it even with 8K images, however I've included this even if I disagree for now. But I can't be sure without knowing the other thread that keeps the lock. One issue with this, is that first frame can be != 0 timestamp. Possible a better way would be to check if the input stream is an image or has total frames < 2.

PS: This function will return false in this case and the Screamer will increase the dropped frames.

**Comment by signumnova** [comment]
Cheers - yes I agree a test for a static image is possibly a better condition for overriding the lock timeout. We can revisit this in the future?
